### DISCUSSION TITLE
Looking for an way to get the traces from 3rd party libraries (Java) when opentelemetry api and sdk are used

### DISCUSSION THREAD

**Discussion by SubhadeepSen** [open-telemetry/opentelemetry-java-instrumentation#3104] [Q&A]
I have been trying to use the opentelemetry-api and opentelemetry-sdk along with the sdk-autoconfigure in a micronaut application to generate the traces. The traces are coming as expected when the application is instrumented. I'm looking for a way to get the traces from 3rd party libraries as well with my current configuration and without using the opentelemetry java agent. I've tried to use the supported libraries as mentioned here (https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md) but I'm not getting any traces from these 3rd party libraries.

Is there any way to get the traces from 3rd party libraries (aws, apache httpclient etc.) when opentelemetry api and sdk are used?

Dependencies:
opentelemetry-api:1.2.0
opentelemetry-sdk:1.2.0
opentelemetry-exporter-otlp:1.2.0
opentelemetry-sdk-extension-autoconfigure:1.2.0-alpha

software.amazon.awssdk:s3:2.15.15
org.apache.httpcomponents:httpclient:4.5

Appreciate any suggestion or guidance.

**Comment by anuraaga** [comment]
Hi @SubhadeepSen - I've transferred this issue to the instrumentation repo since it's about instrumentation.

Unfortunately we have not augmented that list with a checkbox on which libraries we support without the agent, it's only a subset. For now, the best way is to search for a `library` folder for an instrumentation in our repo, such as

https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/grpc-1.6/library

There should be instructions on how to enable the library instrumentation (no agent so it can't happen automatically naturally). If there are any libraries where we don't have a `library` folder that you are looking for, feel free to file an issue and we can try to prioritize it.

**Reply by SubhadeepSen** [reply]
Hi @anuraaga Thank you for responding. Could you please guide me how can I use this aws library (https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/aws-sdk/aws-sdk-2.2/library) and the trace propagation section says that "This format is the only format recognized by AWS managed services". 

Is there any support that the generated traces can be understood by opentelemetry collector ?

**Reply by anuraaga** [reply]
The format is only about propagation, not exporting. If you have a trace _through_ AWS services, e.g. you publish a message to SQS which is handled by Lambda, it will be propagated all the way to Lambda via this mechanism. If you don't have such workloads you won't be affected.

The traces themselves are just collected using OpenTelemetry as normal and can be exported by whatever you configure.

**Reply by SubhadeepSen** [reply]
Could you please help me, how aws-sdk-2.2 can be added as gradle dependency (group and artifact)
